app/controllers/courses_controller.rb

```diff
@@ -1,16 +1,25 @@
 class CoursesController < InheritedResources::Base
   load_and_authorize_resource
 
+  def create
+    @course = Course.new(course_params)
+    if @course.save
+      redirect_to courses_path, notice: 'Course saved.'
+    else
+      render :new
+    end
+  end
+
```

Comment:
`load_and_authorize_resource` is now wrapped in the `cancancan` gem. Controllers have been updated accordingly.